index.js

```diff
@@ -0,0 +1,327 @@
+      reportStatus(`Encountered an error: ${error.message}.`);
+    });
+
+  $('input[name="name"]').val('');
+  $('#continent option:selected').val('');
+  $('input[name="category"]').val('');
+  $('input[name="weeks"]').val('');
+  $('input[name="cost"]').val('');
+  $('#about').val('');
+}
+
+const showSearchTripsForm = () => {
+  reportStatus('');
+  $('#search-trips-form').show();
+  const searchTripsForm = $('#search-trips-form');
+  searchTripsForm.empty();
+
+  searchTripsForm.append(
+    `<h2>Search Trips</h2>`
+  );
+  searchTripsForm.append(
+    `<div>
+      <label class="search-form">Continent:</label>
+      <select id="search-continent" class="search-form">
+        <option></option>
+        <option value="Africa">Africa</option>
+        <option value="Asia">Asia</option>
+        <option value="Australasia">Australasia</option>
+        <option value="Europe">Europe</option>
+        <option value="North America">North America</option>
+        <option value="South America">South America</option>
+      </select>
+    </div>`
+  );
+  searchTripsForm.append(
+    `<div>
+      <label class="search-form">Max Amount of Weeks:</label>
+      <input id="max-weeks" type="number" name="max-weeks" class="search-form" />
+    </div>`
+  );
+  searchTripsForm.append(
+    `<div>
+      <label class="search-form">Max Budget($):</label>
+      <input id="budget" type="number" name="budget" class="search-form" />
+    </div>`
+  );
+  searchTripsForm.append(
+    `<input type="submit" name="search" value="Search" class="button" id="search-by-budget"/>`
+  );
+}
+
+const searchByBudget = () => {
+  reportStatus('');
+  setupTripsView('Search Results', 'Searching');
+  const tripList = $('#trip-list');
+
+  if ($('#budget').val()) {
+    let queryString = '/budget?query=' + $('#budget').val();
+    axios.get(URL + queryString)
+      .then((response) => {
+        if (response.data) {
+          filterAndShowData(response.data, tripList);
+        } else {
+          reportStatus('No Record Found!');
+        }
+      })
+      .catch((error) => {
+        reportStatus(`Error: ${error.message}`);
+      });
+  } else {
+    axios.get(URL)
+      .then((response) => {
+        if (response.data) {
+          filterAndShowData(response.data, tripList);
+        } else {
+          reportStatus('No Record Found!');
+        }
+      })
+      .catch((error) => {
+        reportStatus(`Error: ${error.message}`);
+      });
+  }
+}
+
+// listeners
+$(document).ready(() => {
+  $('#load').click(loadTrips);
+  $('#trips').on('click', 'li', function() {
+    let id = $(this).attr('id');
+    loadTrip(id);
+    window.scrollTo(0, 0);
+  });
+  $('#reservation-form').on('click', '.reserve', function(){
+    let id = $(this).attr('id').substr(3);
+    reserveTrip(id);
+    window.scrollTo(0, 0);
+  });
+
+  $('#search').click(showSearchTripsForm);
+  $('#search-trips-form').on('click', '#search-by-budget', function(){
+    searchByBudget();
+  });
```

Comment:
Since all this anonymous function is doing is calling `searchByBudget()`, you can probably just pass this function in as the callback itself:

```javascript
$('#search-trips-form').on('click', '#search-by-budget', searchByBudget);